### DISCUSSION TITLE
How to create a sample over line on a multiblock object

### DISCUSSION THREAD

**Discussion by npillardou** [pyvista/pyvista#5302] [Q&A] [answered]
Dear community, 

I would like to know if it would be possible to do a sample over line method applied to a multiblock object ? 

I can't find a way to convert my multiblock object to a single object and then to apply this method.

I have tried to wrap or merge it but something went wrong.

```python
file = "vtkOutput.pvd"
dataset = pv.read(file)
finalblock = dataset.GetBlock(0)
tmp = pv.UnstructuredGrid()
for i in range(finalblock.n_blocks):
     block = finalblock.GetBlock(i)
     tmp = pv.wrap(block)

a = [200, 0, 0] 
b = [200, 670, 0]
line = pv.Line(a, b)

dataset_on_line = tmp.sample_over_line(a, b, resolution=1000)
``` 

As my blocks are coming from a partitioned mesh, when I try to plot the values intersected by the line, it only gives me the last values of the last block intersection with the line.

Could you help me to figure out what is missing please ?

Thank you in advance.

**Comment by MatthewFlamm** [comment]
If this filter doesn't work on the `MultiBlock` object itself, i.e. `dataset.sample_over_line` this is a good feature request as I believe it should be possible to implement.

**Reply by MatthewFlamm** [reply, marked as answer]
In the meantime, a possible workaround is to merge the `MultiBlock` mesh first.  Another workaround is to run the sample process iteratively inside your loop. Then combine the result.

As an aside, you can also make your code more pythonic like this:


```py
file = "vtkOutput.pvd"
dataset = pv.read(file)
finalblock = dataset[0]  # Access like a sequence

for block in dataset:  # iterate over the blocks
     # do something with block which is already a PyVista object
   
```
